assets/app/scripts/directives/oscResourceNameValidator.js

```diff
@@ -6,7 +6,8 @@ angular.module("openshiftConsole")
     //github.com/GoogleCloudPlatform/kubernetes/pkg/util/validation.go
     //limiting to valid service names as LCD
     var maxNameLength = 24;
-    var VALID_NAME_RE = /^[a-z]([-a-z0-9]*[a-z0-9])?/i;
+    var minNameLength = 2;
+    var VALID_NAME_RE = /^[a-z]([-a-z0-9]*[a-z0-9])?$/;
```

Comment:
these requirements can't be generalized to all resource types... some allow upper-case, some allow single-character names